tutor/src/components/terms-modal.jsx

```diff
@@ -32,8 +32,11 @@ class TermsModal extends React.Component {
 
   // for terms to be displayed the user must be in a course and need them signed
   @computed get isReady() {
-    return !!(
-      this.props.canBeDisplayed && User.terms_signatures_needed && !isEmpty(User.unsignedTerms)
+    return Boolean(
+      !User.terms.api.isPending &&
+        this.props.canBeDisplayed &&
+        User.terms_signatures_needed &&
+        !isEmpty(User.unsignedTerms)
```

Comment:
We should remove `isEmpty` (and rely on changing `User.terms_signatures_needed`) or also add the isEmpty to the loading screen check.

Reply:
yeah, lets change loading to use that. I think this should still be here because we don't want to display unless there are unsignedTerms